opnode/rollup/driver/state.go

```diff
@@ -469,3 +476,19 @@ func (s *state) loop() {
 		}
 	}
 }
+
+func (s *state) snapshot(event string) {
+	l1HeadJSON, _ := json.Marshal(s.l1Head)
```

Comment:
How do you feel about making a copy and then marshaling / sending in a different thread?

Reply:
I'd rather not complicate things for now. If marshal performance becomes an issue, then we can process these asynchronously.